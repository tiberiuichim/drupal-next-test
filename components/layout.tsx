import { Roboto } from "next/font/google";
import Link from "next/link";

import { Breadcrumb } from "antd";
import { ConfigProvider } from "antd";
import { PreviewAlert } from "components/preview-alert";

import colors from "colors";

const roboto = Roboto({
  weight: ["400", "500", "700"],
  subsets: ["latin"],
  display: "swap",
});

export function Layout({ children }) {
  return (
    <>
      <ConfigProvider
        theme={{
          token: {
            fontSize: 16,
            borderRadius: 0,
            fontFamily: roboto.style.fontFamily,
            colorPrimary: colors["violet-light"],
          },
        }}
      >
        <PreviewAlert />
        <div className="max-w-screen-md px-6 mx-auto">
          <header>
            <div className="container flex items-center justify-between py-6 mx-auto">
              <Breadcrumb items={[{ title: "Home", href: "/" }]} />
              <Link href="/" className="text-2xl font-semibold no-underline">
                Next.js for Drupal
              </Link>
              <Link
                href="https://next-drupal.org/docs"
                target="_blank"
                rel="external"
                className="hover:text-blue-600"
              >
                Read the docs
              </Link>
            </div>
          </header>
          <main className="container py-10 mx-auto">{children}</main>
        </div>
      </ConfigProvider>
    </>
  );
}
