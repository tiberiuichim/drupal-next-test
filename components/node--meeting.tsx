import { DrupalNode } from "next-drupal"

interface NodeMeetingProps {
  node: DrupalNode
}

export function NodeMeeting({ node, ...props }: NodeMeetingProps) {
  return (
    <article {...props}>
      <h1 className="mb-4 text-6xl font-black leading-tight">{node.title}</h1>
      {node.body?.processed && (
        <div
          dangerouslySetInnerHTML={{ __html: node.body?.processed }}
          className="mt-6 font-serif text-xl leading-loose prose"
        />
      )}
    </article>
  )
}
