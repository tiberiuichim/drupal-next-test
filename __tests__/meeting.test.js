import { render } from "@testing-library/react";

import { NodeMeeting } from "components/node--meeting";
import { meetingNode } from "./fixtures";

test("Index page matches snapshot", () => {
  const { asFragment } = render(<NodeMeeting node={meetingNode} />);
  expect(asFragment()).toMatchSnapshot();
});
