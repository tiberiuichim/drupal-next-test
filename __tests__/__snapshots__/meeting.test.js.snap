// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<DocumentFragment>
  <article>
    <h1
      class="mb-4 text-6xl font-black leading-tight"
    >
      88th meeting
    </h1>
  </article>
</DocumentFragment>
`;
